sql/parse/parse.go

```diff
@@ -1360,6 +1372,31 @@ func convertDeclareCursor(ctx *sql.Context, d *sqlparser.Declare) (sql.Node, err
 	return plan.NewDeclareCursor(dCursor.Name, selectStmt), nil
 }
 
+func convertDeclareHandler(ctx *sql.Context, d *sqlparser.Declare, query string) (sql.Node, error) {
+	dHandler := d.Handler
+	//TODO: support other condition values besides NOT FOUND
+	if len(dHandler.ConditionValues) != 1 || dHandler.ConditionValues[0].ValueType != sqlparser.DeclareHandlerCondition_NotFound {
+		return nil, sql.ErrUnsupportedSyntax.New(sqlparser.String(d))
+	}
+	stmt, err := convert(ctx, dHandler.Statement, query)
+	if err != nil {
+		return nil, err
+	}
+
+	var action plan.DeclareHandlerAction
+	switch dHandler.Action {
+	case sqlparser.DeclareHandlerAction_Continue:
+		action = plan.DeclareHandlerAction_Continue
+	case sqlparser.DeclareHandlerAction_Exit:
+		action = plan.DeclareHandlerAction_Exit
+	case sqlparser.DeclareHandlerAction_Undo:
+		action = plan.DeclareHandlerAction_Undo
+	default:
+		return nil, fmt.Errorf("unknown DECLARE ... HANDLER action: %v", dHandler.Action)
```

Comment:
Shouldn't this be unsupported feature err?

Reply:
Nope, as there should only be the three listed. If there's another one, then there shouldn't be and it's unknown.